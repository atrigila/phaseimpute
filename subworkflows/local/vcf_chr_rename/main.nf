include { BCFTOOLS_ANNOTATE           } from '../../../modules/nf-core/bcftools/annotate/main.nf'
include { BCFTOOLS_INDEX as VCF_INDEX } from '../../../modules/nf-core/bcftools/index/main.nf'

workflow VCF_CHR_RENAME {
    take:
    ch_vcf          // channel: [ [id, ref], vcf, csi ]
    file_chr_rename // file

    main:

    ch_versions = Channel.empty()

    // Rename the chromosome without prefix
    BCFTOOLS_ANNOTATE(ch_vcf
        .combine(Channel.of([[], [], []]))
        .combine(Channel.of(file_chr_rename))
    )
    
    VCF_INDEX(BCFTOOLS_ANNOTATE.out.vcf)
    ch_versions = ch_versions.mix(VCF_INDEX.out.versions.first())

    ch_vcf_rename = BCFTOOLS_ANNOTATE.out.vcf
        .combine(VCF_INDEX.out.csi)

    emit:
    vcf_rename     = ch_vcf_rename         // [ meta, vcf, csi ]
    versions       = ch_versions           // channel: [ versions.yml ]
}