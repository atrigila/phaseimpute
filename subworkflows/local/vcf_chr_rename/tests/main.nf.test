nextflow_workflow {

    name "Test Subworkflow VCF_CHR_RENAME"
    script "../main.nf"
    config "./nextflow.config"

    workflow "VCF_CHR_RENAME"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "subworkflows/vcf_chr_rename"
    tag "vcf_chr_rename"

    tag "bcftools"
    tag "bcftools/annotate"
    tag "bcftools/index"

    test("Should run without error") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id: "input", genome:"GRCh37"],
                    "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/vcf/NA24385_sv.vcf.gz",
                    "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/vcf/NA24385_sv.vcf.gz.tbi"
                ])
                input[1] = file("../../../assets/chr_rename_add.txt", exist: true)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.ch_region).match() }
            )
        }
    }
}
