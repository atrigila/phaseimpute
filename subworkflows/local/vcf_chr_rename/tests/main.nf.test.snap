{
    "Should run without error": {
        "content": null,
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-13T12:47:49.775995"
    }
}